### ISSUE TITLE
Internal error: pc read in psymtab, but not in symtab when debugging with GDB

### ISSUE THREAD

**Issue by Geod24** [ldc-developers/ldc#3905]
We have a [fairly large project](https://github.com/bosagora/agora) with many dependencies, including Vibe.d.
Recently we started to distribute binaries for Ubuntu 20.04 x86_64, built on the Github runners.
That latest LDC (v1.28.1) is downloaded from `dlang-community/setup-dlang`, which AFAIK uses the releases from this repository.

Our [latest release](https://github.com/bosagora/agora/releases/tag/v0.32.0) had a silly `null` pointer dereference which caused a SEGV. We got the `core` file from the servers (Ubuntu 20.04) and entered it in `gdb`:
```
# gdb --version
GNU gdb (Ubuntu 9.2-0ubuntu1~20.04.1) 9.2
Copyright (C) 2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

This is what the `bt` output looks like:
<img width="1680" alt="Screen Shot 2022-01-20 at 18 42 04" src="https://user-images.githubusercontent.com/2180215/150313407-576d0ccd-8490-44f7-b72d-a56bbf1dfa7a.png">

The build log for the binary is [here](https://github.com/bosagora/agora/runs/4865242833?check_suite_focus=true).
Something that might be relevant, we use `--export-dynamic` as it allows us to have better backtrace (normally).

I have also attached the [core dump](https://github.com/ldc-developers/ldc/files/7904198/core.zip): It's 4.3G uncompressed, but the `zip` contains an `lz4` archive (because Github wants `zip` and `systemd` gave me `lz4`).

**Comment by kinke** [comment]
Have you tried it without `--export-dynamic`? IIRC, they're not needed for gdb backtraces, just for the druntime one.

**Comment by Geod24** [comment]
Yes, same problem. BTW in our case, we use `--export-dynamic` for Tracy, too.
